src/Settings.py

```diff
@@ -74,3 +74,43 @@ def get_username(capitalize=False):
             username = username.capitalize()
 
         return username
+
+    def reset_discord_endpoints():
+        Settings.set_key(Settings.discord_prod_key, "")
+        Settings.set_key(Settings.discord_dev_key, "")
+        Settings.set_discord_endpoints()
+
+    def check_discord_endpoints():
+        settings = Settings.get_all()
+
+        if not Settings.discord_prod_key in settings or \
+           not Settings.discord_dev_key in settings:
+            return False
+
+        if settings[Settings.discord_prod_key] == "" or \
+           settings[Settings.discord_dev_key] == "":
+            return False
+
+        return True
+
+    def set_discord_endpoints():
+        if not Settings.check_discord_endpoints():
+            drive = Drive()
+
+            Settings.set_key(
+                key  = Settings.discord_prod_key,
+                data = drive.get_json_key(
+                    remote_file    = f'{LOFSM_DIR_PATH}/db.json',
+                    local_filepath = f'temp/db.json',
+                    key            = Settings.discord_prod_key
+                )
+            )
+
+            Settings.set_key(
+                key  = Settings.discord_dev_key,
+                data = drive.get_json_key(
+                    remote_file    = f'{LOFSM_DIR_PATH}/db.json',
+                    local_filepath = f'temp/db.json',
+                    key            = Settings.discord_dev_key
+                )
+            )
```

Comment:
@vindennl48 Unfortunately I couldn't figure out how to put this behavior in `Discord.py` like we discussed. I don't see a way to avoid the circular dependency, and I don't think I fully understood how to implement this like you were suggesting. Could you point me in the right direction?

Reply:
For future reference, we found a bug in Settings that it was referencing Settings.get_slack_endpoints and that function doesn't exist.  It was moved to Slack.get_slack_endpoints, and that function was never called to cause an issue.  We will keep it in Settings class for now.